psst-gui/src/webapi/client.rs

```diff
@@ -213,8 +271,280 @@ impl WebApi {
+                            id: id.into(),
+                            name: Arc::from(item.content.data.profile.as_ref().unwrap().name.clone()),
+                            images: item.content.data.visuals.as_ref().map_or_else(Vector::new, |images| 
+                                images.avatar_image.sources.iter().map(|img| data::utils::Image {
+                                    url: Arc::from(img.url.as_str()),
+                                    width: None,
+                                    height: None,
+                                }).collect()
+                            ),
+                        })
+                    },
+                    DataTypename::Album => {
+                        album.push_back(Arc::new(Album {
+                            id: id.into(),
+                            name: Arc::from(item.content.data.name.clone().unwrap()),
+                            album_type: AlbumType::Album,
+                            images: item.content.data.cover_art.as_ref().map_or_else(Vector::new, |images| 
+                                images.sources.iter().map(|src| data::utils::Image {
+                                    url: Arc::from(src.url.clone()),
+                                    width: None,
+                                    height: None,
+                                }).collect()
+                            ),
+                            artists: item.content.data.artists.as_ref().map_or_else(Vector::new, |artists|
+                                artists.items.iter().map(|artist| ArtistLink {
+                                    id: Arc::from(artist.uri.split(':').last().unwrap_or("").to_string()),
+                                    name: Arc::from(artist.profile.name.clone()),
+                                }).collect()),
+                            copyrights: Vector::new(),
+                            label: "".into(),
+                            tracks: Vector::new(),
+                            release_date: None,
+                            release_date_precision: None,
+                        }))
+                    },
+                    DataTypename::Podcast => {
+                        show.push_back(Arc::new(Show {
+                            id: id.into(),
+                            name: Arc::from(item.content.data.name.clone().unwrap()),
+                            images: item.content.data.cover_art.as_ref().map_or_else(Vector::new, |images| 
+                                images.sources.iter().map(|src| data::utils::Image {
+                                    url: Arc::from(src.url.clone()),
+                                    width: None,
+                                    height: None,
+                                }).collect()
+                            ),
+                            publisher: Arc::from(item.content.data.publisher.as_ref().unwrap().name.clone()),
+                            description: "".into(),
```

Comment:
I think we might need to do some truncation of all the descriptions so we can be sure that nothing on the Home page gets too tall from the descriptions.